### ISSUE TITLE
Market doesnt get put out when on fire sometimes

### ISSUE THREAD

**Issue by Monsterfisch** [Sh0wdown/UnofficialCrusaderPatch#687]
I am not sure if its a regular thing or just affects some people/versions of the game but for me, in the 1.14.1 shc extreme version markets that get set on fire(for the ai and bay the ai, I haven't tested it for the player) the market sometimes doesn't get put out by any fireman and keeps on burning until it's destroyed. potentially setting fire to more buildings around it unless they are still immune from fire after being put out previously

I am not quite sure how to replicate it as it does sort of happening randomly =/

**Comment by LordHansCapon** [comment]
The building type does not matter when putting out the fire. So I guess it can happen to any flammable building I suppose. The worker AI works the same for every player, so that can't be a factor. I couldnt find anything bad related to this part of the code, so it requires a more in-depth investigation to get to the bottom of it.

**Comment by GRhin** [comment]
Just a random thought, unlikely but possible.
Do the firemen prioritize buildings based on cost? Maybe the 0 cost marketplace causes the issue.

**Comment by Monsterfisch** [comment]
I raised that idea with showdown in the past as well as the "bug" became apparent to me around the time the marketplace price reduction was introduced but I haven't done enough testing to confirm that it is related. I can try to get a save file where it has the market burning and doesn't get put out but I am somewhat certain that it doesn't stay that way after loading the map

**Comment by Heroesflorian** [comment]
> Do the firemen prioritize buildings based on cost? Maybe the 0 cost marketplace causes the issue.
No, I can confirm this thing with reachable trading posts not getting extinguished also happened relatively frequently for some custom Saladin AIV I had created and watched in action before the UCP, let alone 0-cost trading posts were introduced to the game. 
Also, well workers do happily run for a 5 wood apple farm when there's a cathedral and a quarry burning elsewhere as well, so I doubt they have any idea of building cost influencing their decisions anyway? Haven't made further tests on that, though, but as I said, the issue was there before free markets ever were possible.

**Comment by Monsterfisch** [comment]
In this savefile, I have "captured" a market that is burning and getting ignored by the fireman at the red wazir hopefully its helpfull =)

**Comment by Krarilotus** [comment]
as far as i am concerned, i have also seeen a granary not getting put out. It could be related to the AI not storingthe first buildings it sets up in some matrix for buildigns to care for their state or sth
